### PR TITLE
Update to directory structure

### PR DESCRIPTION
**Pull request by katjuell**
Resolves [this Asana](https://app.asana.com/0/1200198213175654/1200638573421026/f) by removing the redirect in our top-level Next config and the `posts/index` page.

For the redirect: If we have the dynamic slug template at the top level of the site, we don't need a redirect in the Next config — Next will work within the existing directory structure: https://nextjs.org/docs/routing/introduction. Any requests for cached routes we can handle at the proxy level, which is what Netlify recommends. More specifically, they claim that redirects in `next.config.js` won't work: "Redirects and rewrites using next.config.js aren’t currently supported for Next.js sites on Netlify." https://docs.netlify.com/configure-builds/common-configurations/next-js/. Instead, they recommend adding a `_redirects` file to the publish directory, which we generate dynamically for each deploy, or setting redirects in a `netlify.toml` file at the project root, which this PR does. NB: the default in the config is a `301` & `force = true`, which is what we had in the Next config, so I assume this is what we want.

For the removal of `posts/index`: Our landing page currently functions as an index for posts, so this page seems redundant currently. If we want to ensure that we have a top-level landing page for tags, then it seems as though adding a tags index page would make the most sense (seems like it should be a separate PR).

## To Test
- Check out the branch & make sure that the site still behaves as you would expect from the main `index`.
- Check that `/posts` gets a 404. **Though we could also add a redirect back to `/`**
- Check the deploy preview and ensure that a given post redirects correctly. So `/posts/how-to-install-sourcegraph-with-docker` should redirect to `/how-to-install-sourcegraph-with-docker`

### CODE REVIEW

File: pages/[slug].tsx [hunk 1]
```diff
@@ -44,3 +42,5 @@ export const getStaticProps: GetStaticProps<ArticleProps> = async context => {
         }),
     }
 }
+
+export default Article
```
Comment (katjuell) [inline pages/[slug].tsx:46]: Not related to this PR, just something I noticed/updated for consistency.

**Comment by katjuell** [comment]
@ltagliaferri @marekweb I kind of buried my questions a bit in the PR message, but I think to sum up:
- Do we want a redirect for the posts index page since we are removing it? My guess is that as we add more content types, we will bring a posts index page back, so I would personally be okay not doing the redirect for now, since I doubt it was widely indexed.
- Do we want a tags index page? High/low/medium priority?

**Comment by marekweb** [comment]
> Do we want a redirect for the posts index page since we are removing it? My guess is that as we add more content types, we will bring a posts index page back, so I would personally be okay not doing the redirect for now, since I doubt it was widely indexed.

We didn't share that page link anywhere, so I agree that we don't need to worry about a redirect.

> Do we want a tags index page? High/low/medium priority?

I think it's relatively low priority for the same reason as stated above, namely that our landing page is sufficient to index all of our posts for now, but we should revisit this when we have more content.

**Comment by katjuell** [comment]
That sounds good to me! This should be good to go then.

**Review by ltagliaferri — APPROVED** [review]
Yes, agree with all of Marek's comments here. Tags index would be a nice to have but probably not a high priority while we are roadmapping.

**Pull request merged**